src/traces/scatter3d/convert.js

```diff
@@ -345,6 +345,20 @@ function arrayToColor(color) {
     return null;
 }
 
+function arrayToColor(colors) {
+    if(!Lib.isArrayOrTypedArray(colors)) {
+        return _arrayToColor(colors);
```

Comment:
`return null` here and avoid the outer `isArrayOrTypedArray` test in `_arrayToColor`?

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4867/commits/33fd07d9dc501d24b6be4e426918b6bb0dcd9a5f.